transfer_nlp/plugins/trainers.py

```diff
@@ -219,6 +213,109 @@ def log_test_results(trainer):
             store_metrics(metrics=metrics, mode="test")
             logger.info(f"Test Results - Epoch: {trainer.state.epoch} {print_metrics(metrics)}")
 
+    def custom_setup(self):
+        raise NotImplementedError
```

Comment:
should this be an `ABC` then? also these methods should have docstrings, and it's odd this method is declared here but only actually used in SingleTaskTrainer

Reply:
is this method really needed? child classes can just do stuff in their own `__init__` method?

Reply:
True, I think this can be removed from this base class